cs-minibot-platform-src/src/main/java/simulator/simbot/Dijkstras.java

```diff
@@ -0,0 +1,199 @@
+    public static Node[][] initializeNodeMatrix(int[][] maze) {
+        int height = maze.length;
+        int width = maze[0].length;
+        Node[][] res = new Node[height][width];
+        for(int i = 0; i < height; i++) {
+            for(int j = 0; j < width; j++) {
+                Node n = new Node(maze[i][j], false,i,j);
+                if(i == height-2 && j == width-2) {
+                    targetNode = n;
+                }
+                res[i][j] = n;
+                if(maze[i][j] == 0) {
+                    res[i][j].cost = Integer.MAX_VALUE;
+                }
+            }
+        }
+
+        res[height-2][width-2].isEnd = true;
+        res[1][1].cost = 1;
+        return res;
+    }
+    //Returns a HashSet of neighbors given a position and maze
+    public static HashSet<Node> generateNeighbors(int i, int j, Node[][] maze) {
+        HashSet<Node> res = new HashSet<Node>();
+        if(maze[i+1][j].value == 0) {
+            res.add(maze[i+1][j]);
+        }
+        if(maze[i][j+1].value == 0) {
+            res.add(maze[i][j+1]);
+        }
+        if(maze[i-1][j].value == 0) {
+            res.add(maze[i-1][j]);
+        }
+        if(maze[i][j-1].value == 0) {
+            res.add(maze[i][j-1]);
+        }
+        return res;
+    }
+
+    //Returns an array of Nodes with the neighbors field filled out
+    public static Node[][] populateNeighbors(Node[][] maze) {
+        int height = maze.length;
+        int width = maze[0].length;
+        for(int i = 1; i < height-1; i++) {
+            for(int j = 1; j < width -1; j++) {
+                if(maze[i][j].value == 0) {
+                    maze[i][j].neighbors = generateNeighbors(i, j, maze);
+                }
+            }
+        }
+        return maze;
+    }
+
+    public static Node getNodeWithLowestCost(HashSet<Node> unsettledNodes) {
+        int lowestCost = Integer.MAX_VALUE;
+        Node res = null;
+        for(Node n: unsettledNodes) {
+            if(n.cost < lowestCost) {
+                lowestCost = n.cost;
+                res = n;
+
+            }
+        }
+        return res;
+    }
+
+    public static void processNeighbors(Node currentNode) {
+        for(Node n: currentNode.neighbors) {
+            if(!settledNodes.contains(n)) {
+                int newCost = currentNode.cost + 1;
+                if(newCost < n.cost) {
+
+                    n.cost = newCost;
+                    predecessors.put(n, currentNode);
+                }
+                unsettledNodes.add(n);
+            }
+        }
+    }
+    public static LinkedList<Node> getPath(Node target) {
+        LinkedList<Node> path = new LinkedList<Node>();
+        Node step = target;
+        // check if a path exists
+        if (predecessors.get(step) == null) {
+            return null;
+        }
+        path.add(step);
+        while (predecessors.get(step) != null) {
+            step = predecessors.get(step);
+            path.add(step);
+
+
```

Comment:
fix these newlines- I think it would be easiest to just right click the file in intellij and click reformat code.